src/FffGcodeWriter.cpp

```diff
@@ -712,15 +712,39 @@ LayerPlan& FffGcodeWriter::processLayer(const SliceDataStorage& storage, int lay
     return gcode_layer;
 }
 
-bool FffGcodeWriter::getExtrudersNeedPrimeDuringFirstLayer() const
+bool FffGcodeWriter::getAnyExtruderNeedPrimeDuringFirstLayer(const SliceDataStorage& storage) const
```

Comment:
Why not make a function `getExtruderNeedsPrimeDuringFirstLayer` which works per extruder?

Nothing depends on whether any extruder needs a prime; stuff only depends per extruder on whether a prime poop is needed.